File: java/src/no/sindreboyum/fizzbuzz/FizzBuzz.java
package no.sindreboyum.fizzbuzz;

public class FizzBuzz {
    public static void main(String args[]) {
        

        int i = 1;
        int n = 100;
        String result = "";

        boolean b;
        boolean f;

        for (; i <= n; i++, result = "") {
            b = i % 3 == 0;
            f = i % 5 == 0;

            result += b ? "fizz" : "";
            result += f ? "buzz" : "";

            if (!(b && f)) {
                result += i;
            }

            System.out.println(result);
        }
    }
}

File: f#/fizzbuzz.fs
open System

let numbers = [| for i in 1 .. 100 ->  i |]
let isFizz number = number % 3 = 0
let isBuzz number = number % 5 = 0

let fizzorbuzz number = 
  if isFizz number && isBuzz number 
  then Console.WriteLine "FizzBuzz" 
  else 
  (
    if isFizz number 
    then Console.WriteLine "Fizz" 
    else 
    (
      if isBuzz number 
      then Console.WriteLine "Buzz" 
      else Console.WriteLine (string number)
    )
  )

let fizzbuzz numbers = Array.iter fizzorbuzz numbers

fizzbuzz numbers
